addons/l10n_ar/models/res_partner.py

```diff
@@ -129,8 +129,9 @@ def _get_id_number_sanitize(self):
             return 0
         if self.l10n_latam_identification_type_id.l10n_ar_afip_code in ['80', '86']:
             # Compact is the number clean up, remove all separators leave only digits
-            res = int(stdnum.ar.cuit.compact(self.vat))
+            compacted = stdnum.ar.cuit.compact(self.vat)
+            res = int(compacted) if compacted.isdigit() else 0
```

Comment:
Instead of this change, I would prefer to return 0 if VAT is not valid by using `_check_vat_number` at staring of this method
```py
if not (self.vat and self._check_vat_number('AR', self.vat)):
    return 0
```